beets/importer.py

```diff
@@ -943,7 +943,9 @@ def is_archive(cls, path):
         if not os.path.isfile(path):
             return False
 
-        for path_test, _ in cls.handlers():
+        for path_test, x in cls.handlers():
+            if isinstance(path, bytes):
+                path = util.py3_path(path)
```

Comment:
Can we test that assumption? If it's true, maybe we should just convert the results from those Unicode-only APIs back to our uniform representation (bytes).